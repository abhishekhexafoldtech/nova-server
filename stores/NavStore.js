import { defineStore } from "pinia";
import bootstrap from "bootstrap/dist/js/bootstrap.min.js";

export const useNavStore = defineStore("NavStore", {
  // state
  state: () => {
    return {
      bootstrap: bootstrap,
      isSidenavCollapsed: false,
      showConfig: false,
      sidenavType: 'bg-white',
      navFixed: false,
      darkMode: false,
      miniNotification: false,
      breadCrumb: "Nova-Gas"
    };
  },

  actions: {
    toggleSidenavCollapsing() {
      this.isSidenavCollapsed = !this.isSidenavCollapsed;
    },
    toggleShowConfig() {
      this.showConfig = !this.showConfig;
    },
    toggleSidebarColor(payload) {
      this.sidebarColor = payload;
    },
    setSidenavType(payload) {
      this.sidenavType = payload;
    },
    toggleDarkMode() {
      this.darkMode = !this.darkMode;
    },
    toggleNavFixed() {
      this.navFixed = !this.navFixed;
    },
    setBreadCrumbs(data) {
      this.breadCrumb = data
    }
  },
});

