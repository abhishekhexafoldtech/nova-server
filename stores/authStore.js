import { defineStore } from 'pinia';
import { getToken, removeToken, setToken } from '@/utils/auth'
import {  getUserInfo, loginByWeb, logoutByWeb } from '@/api/login'

export const useAuthStore = defineStore("auth", {

  state: () => ({
    token: getToken(),
    user_info: {}, // complete user details
    page_refreshing: true,
  }),

  getters: {
    getLoginToken() {
      return this.token
    },
  },

  actions: {
	// Login
	Login(userCredInfo) {
		return new Promise((resolve, reject) => {
			// make login data set
			let data = {
				email: userCredInfo.email,
				phone: userCredInfo.phone,
				password: userCredInfo.password,
			}
			// Lets fire login with info
			loginByWeb(data).then(response => {
				if(response.status == 200) {
					let responseData = response.data
					// console.log("responseData", responseData)
					let go = true
						if (go) {
							// SITE TOKEN
							this.token = responseData.access_token
							this.user_info = responseData.data
							setToken(responseData.access_token)

							const person = { 
									"id": responseData.data.id,
									"first_name": responseData.data.first_name,
									"last_name": responseData.data.last_name,
									"email": responseData.data.email,
									"phone": responseData.data.phone,
								}
							
							// console.log('person', person)
							localStorage.setItem("user", JSON.stringify(person));

							resolve()
						}
					}
						else {
						alert('Something went wrong!')
						reject()
					}
			}).catch(error => {
				reject(error)
			})
		})
	},

	// Logout
	LogOut() {
		return new Promise((resolve, reject) => {
			this.token = ""
			removeToken()
			localStorage.clear();
			resolve()
			// logoutByWeb().then((response) => {
			// 	this.token = ""
			// 	removeToken()
			// 	resolve(response)
			// }).catch(error => {
			// 	console.log(error)
			// 	reject(error)
			// })
			// .finally(() => {
			// })
		})
	},

	// Navbar loaded or refreshed : both cases fired
	pageLoadedORRefreshed() {
		// Get user info
		return new Promise((resolve) => {
			this.page_refreshing = true;
			// Get user info
			// getUserInfo().then((response) => {
			// this.user_info = response.data.data
			// 	resolve(response);
			// })
			// Get user info from local storage
			let temp_data = localStorage.getItem("user")
			this.user_info = JSON.parse(temp_data)
		});
	},
  },
});
