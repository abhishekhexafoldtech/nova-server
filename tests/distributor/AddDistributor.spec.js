import { describe, it, expect } from "vitest";
import { mount, shallowMount } from "@vue/test-utils";
import AddDistributor from "@/pages/distributor/add-distributor/index.vue";
import { pinia } from "@/tests/mockpinia";
import { useNavStore } from "@/stores/NavStore";

describe("AddNewDistributor", () => {
  const navStore = useNavStore(pinia);
  it("renders all input fields", () => {
    const wrapper =shallowMount(AddDistributor);

    // Check if the First Name input field is present
    expect(wrapper.find('[placeholder="First name"]').exists()).toBe(true);

    // Check if the Last Name input field is present
    expect(wrapper.find('[placeholder="Last name"]').exists()).toBe(true);

    // Check if the Phone Number input field is present
    expect(wrapper.find('[placeholder="Phone number"]').exists()).toBe(true);

    // Check if the Email input field is present
    expect(wrapper.find('[placeholder="Email"]').exists()).toBe(true);

    expect(wrapper.html()).toMatchSnapshot();
  });

  it("renders the Continue button", () => {
    const wrapper = shallowMount(AddDistributor);

    // Check if the "Continue" button is present
    expect(wrapper.find(".btn-primary").text()).toBe("Continue");
  });
});
