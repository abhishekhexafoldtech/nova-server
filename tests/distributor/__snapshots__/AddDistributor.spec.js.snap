// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AddNewDistributor > renders all input fields 1`] = `
"<section data-v-ff0b6a8d=\\"\\" class=\\"edit_mang_wrap\\">
  <div data-v-ff0b6a8d=\\"\\" class=\\"mang_inner add_admin\\">
    <div data-v-ff0b6a8d=\\"\\" class=\\"mang_title\\">
      <h3 data-v-ff0b6a8d=\\"\\">Add distributor</h3>
      <p data-v-ff0b6a8d=\\"\\">Enter your details</p>
    </div><!-- form -->
    <el-form data-v-ff0b6a8d=\\"\\" model=\\"[object Object]\\" rules=\\"[object Object]\\" label-position=\\"top\\" label-width=\\"100px\\" size=\\"large\\">
      <el-row data-v-ff0b6a8d=\\"\\" gutter=\\"25\\">
        <!-- First Name -->
        <el-col data-v-ff0b6a8d=\\"\\" xs=\\"24\\" sm=\\"12\\" md=\\"12\\" lg=\\"12\\" xl=\\"12\\">
          <div data-v-ff0b6a8d=\\"\\" class=\\"fieldrow\\">
            <el-form-item data-v-ff0b6a8d=\\"\\" label=\\"First Name\\" prop=\\"first_name\\">
              <el-input data-v-ff0b6a8d=\\"\\" placeholder=\\"First name\\" modelvalue=\\"\\"></el-input>
            </el-form-item>
          </div>
        </el-col><!-- Last Name -->
        <el-col data-v-ff0b6a8d=\\"\\" xs=\\"24\\" sm=\\"12\\" md=\\"12\\" lg=\\"12\\" xl=\\"12\\">
          <div data-v-ff0b6a8d=\\"\\" class=\\"fieldrow\\">
            <el-form-item data-v-ff0b6a8d=\\"\\" label=\\"Last Name\\" prop=\\"last_name\\">
              <el-input data-v-ff0b6a8d=\\"\\" placeholder=\\"Last name\\" modelvalue=\\"\\"></el-input>
            </el-form-item>
          </div>
        </el-col><!-- Phone Number -->
        <el-col data-v-ff0b6a8d=\\"\\" xs=\\"24\\" sm=\\"12\\" md=\\"12\\" lg=\\"12\\" xl=\\"12\\">
          <div data-v-ff0b6a8d=\\"\\" class=\\"fieldrow\\">
            <el-form-item data-v-ff0b6a8d=\\"\\" label=\\"Phone Number\\" prop=\\"phone\\">
              <el-input data-v-ff0b6a8d=\\"\\" type=\\"number\\" placeholder=\\"Phone number\\" modelvalue=\\"\\"></el-input>
            </el-form-item>
          </div>
        </el-col><!-- Email -->
        <el-col data-v-ff0b6a8d=\\"\\" xs=\\"24\\" sm=\\"12\\" md=\\"12\\" lg=\\"12\\" xl=\\"12\\">
          <div data-v-ff0b6a8d=\\"\\" class=\\"fieldrow\\">
            <el-form-item data-v-ff0b6a8d=\\"\\" label=\\"Email\\" prop=\\"email\\">
              <el-input data-v-ff0b6a8d=\\"\\" type=\\"email\\" placeholder=\\"Email\\" modelvalue=\\"\\"></el-input>
            </el-form-item>
          </div>
        </el-col><!-- Role -->
        <!-- <el-col :xs=\\"24\\" :sm=\\"12\\" :md=\\"12\\" :lg=\\"12\\" :xl=\\"12\\">
            <div class=\\"fieldrow\\">
              <el-form-item v-if=\\"newDistributorForm\\" label=\\"Role\\" prop=\\"role\\">
                <el-select
                  class=\\"form_input\\"
                  placeholder=\\"please select role\\"
                  v-model=\\"newDistributorForm.role\\"
                >
                  <el-option label=\\"Admin role\\" value=\\"admin_role\\" />
                  <el-option label=\\"Super Admin Role\\" value=\\"super_admin\\" />
                </el-select>
              </el-form-item>
            </div>
          </el-col> -->
      </el-row>
    </el-form>
    <div data-v-ff0b6a8d=\\"\\" class=\\"perm_footer\\">
      <el-button data-v-ff0b6a8d=\\"\\" class=\\"btn btn-default\\">Cancel</el-button>
      <el-button data-v-ff0b6a8d=\\"\\" class=\\"btn btn-primary\\" type=\\"primary\\">Continue</el-button>
    </div>
  </div>
</section>"
`;

exports[`AddNewDistributor > renders the Continue button 1`] = `
"<section data-v-ff0b6a8d=\\"\\" class=\\"edit_mang_wrap\\">
  <div data-v-ff0b6a8d=\\"\\" class=\\"mang_inner add_admin\\">
    <div data-v-ff0b6a8d=\\"\\" class=\\"mang_title\\">
      <h3 data-v-ff0b6a8d=\\"\\">Add distributor</h3>
      <p data-v-ff0b6a8d=\\"\\">Enter your details</p>
    </div><!-- form -->
    <el-form data-v-ff0b6a8d=\\"\\" model=\\"[object Object]\\" rules=\\"[object Object]\\" label-position=\\"top\\" label-width=\\"100px\\" size=\\"large\\">
      <el-row data-v-ff0b6a8d=\\"\\" gutter=\\"25\\">
        <!-- First Name -->
        <el-col data-v-ff0b6a8d=\\"\\" xs=\\"24\\" sm=\\"12\\" md=\\"12\\" lg=\\"12\\" xl=\\"12\\">
          <div data-v-ff0b6a8d=\\"\\" class=\\"fieldrow\\">
            <el-form-item data-v-ff0b6a8d=\\"\\" label=\\"First Name\\" prop=\\"first_name\\">
              <el-input data-v-ff0b6a8d=\\"\\" placeholder=\\"First name\\" modelvalue=\\"\\"></el-input>
            </el-form-item>
          </div>
        </el-col><!-- Last Name -->
        <el-col data-v-ff0b6a8d=\\"\\" xs=\\"24\\" sm=\\"12\\" md=\\"12\\" lg=\\"12\\" xl=\\"12\\">
          <div data-v-ff0b6a8d=\\"\\" class=\\"fieldrow\\">
            <el-form-item data-v-ff0b6a8d=\\"\\" label=\\"Last Name\\" prop=\\"last_name\\">
              <el-input data-v-ff0b6a8d=\\"\\" placeholder=\\"Last name\\" modelvalue=\\"\\"></el-input>
            </el-form-item>
          </div>
        </el-col><!-- Phone Number -->
        <el-col data-v-ff0b6a8d=\\"\\" xs=\\"24\\" sm=\\"12\\" md=\\"12\\" lg=\\"12\\" xl=\\"12\\">
          <div data-v-ff0b6a8d=\\"\\" class=\\"fieldrow\\">
            <el-form-item data-v-ff0b6a8d=\\"\\" label=\\"Phone Number\\" prop=\\"phone\\">
              <el-input data-v-ff0b6a8d=\\"\\" type=\\"number\\" placeholder=\\"Phone number\\" modelvalue=\\"\\"></el-input>
            </el-form-item>
          </div>
        </el-col><!-- Email -->
        <el-col data-v-ff0b6a8d=\\"\\" xs=\\"24\\" sm=\\"12\\" md=\\"12\\" lg=\\"12\\" xl=\\"12\\">
          <div data-v-ff0b6a8d=\\"\\" class=\\"fieldrow\\">
            <el-form-item data-v-ff0b6a8d=\\"\\" label=\\"Email\\" prop=\\"email\\">
              <el-input data-v-ff0b6a8d=\\"\\" type=\\"email\\" placeholder=\\"Email\\" modelvalue=\\"\\"></el-input>
            </el-form-item>
          </div>
        </el-col><!-- Role -->
        <!-- <el-col :xs=\\"24\\" :sm=\\"12\\" :md=\\"12\\" :lg=\\"12\\" :xl=\\"12\\">
            <div class=\\"fieldrow\\">
              <el-form-item v-if=\\"newDistributorForm\\" label=\\"Role\\" prop=\\"role\\">
                <el-select
                  class=\\"form_input\\"
                  placeholder=\\"please select role\\"
                  v-model=\\"newDistributorForm.role\\"
                >
                  <el-option label=\\"Admin role\\" value=\\"admin_role\\" />
                  <el-option label=\\"Super Admin Role\\" value=\\"super_admin\\" />
                </el-select>
              </el-form-item>
            </div>
          </el-col> -->
      </el-row>
    </el-form>
    <div data-v-ff0b6a8d=\\"\\" class=\\"perm_footer\\">
      <el-button data-v-ff0b6a8d=\\"\\" class=\\"btn btn-default\\">Cancel</el-button>
      <el-button data-v-ff0b6a8d=\\"\\" class=\\"btn btn-primary\\" type=\\"primary\\">Continue</el-button>
    </div>
  </div>
</section>"
`;
